origin-js/src/contractInterface/marketplace/v00_adapter.js

```diff
@@ -241,6 +251,15 @@ class V00_MarkeplaceAdapter {
       fromBlock: this.blockEpoch
     })
 
+    // if an account is passed in, we're looking for the listing state at the time
+    // that account made an offer, so find the block number of the offer
+    // and ignore any "ListingUpdated" events after that block number
```

Comment:
I may not fully get what's going on here, but it seems that it may cause trouble to use `account` for getting old versions of a listing. For example, couldn't an account have multiple offers open on the same listing? E.g. I bought one beanie baby, but then the next day the seller dropped the price so I decided to buy another one. 

Seems more generic and natural to either (a) pass in an offerId to get the listing as it exists for that offer or (b) pass in a block number and let the dapp do its own logic to figure out which block number it wants.